pacbio-subreads-processing.wdl

```diff
@@ -24,172 +24,221 @@ import "structs.wdl" as structs
+            "pacbio-merge": "lumc/pacbio-merge:0.2"
         }
     }
 
     meta {allowNestedInputs: true}
 
-    SubreadsConfig subreadsConfig = read_json(subreadsConfigFile)
-    Array[Subreads] allSubreads = subreadsConfig.subreads
+    # The name of the subreads, to be used to determine output filenames.
+    File subreadsName = basename(subreadsFile, ".subreads.bam")
+
+    # Get the CCS chunks.
+    call ccsChunks as createChunks {
+        input:
+            chunkCount = ccsChunks,
+            dockerImage = dockerImages["python3"]
+    }
+
+    # Index the input bamfile.
+    if (!defined(subreadsIndexFile)) {
+        call pbbam.Index as pbindex {
+            input:
+                bamFile = subreadsFile
+        }
+    }
+
+    File subreadsIndex = select_first([pbindex.index, subreadsIndexFile])
+    File subreadsBam = select_first([pbindex.indexedBam, subreadsFile])
+
+    scatter (chunk in createChunks.chunks) {
+        # Convert chunk from 1/10 to 1 to determine output filename.
+        String chunkNumber = sub(chunk, "/.*$", "")
 
-    scatter (subreads in allSubreads) {
         call ccs.CCS as ccs {
             input:
-                subreadsFile = subreads.subreads_file,
-                outputPrefix = outputDirectory + "/" + subreads.subreads_id + "/" + subreads.subreads_id,
+                subreadsFile = subreadsBam,
+                subreadsIndexFile = subreadsIndex,
+                outputPrefix = subreadsName + "_chunk" + chunkNumber,
                 cores = ccsCores,
+                chunkString = chunk,
                 dockerImage = dockerImages["ccs"]
         }
+    }
 
-        call lima.Lima as lima {
-            input:
-                libraryDesign = libraryDesign,
-                ccsMode = ccsMode,
-                splitBamNamed = splitBamNamed,
-                inputBamFile = ccs.ccsBam,
-                barcodeFile = subreads.barcodes_file,
-                outputPrefix = outputDirectory + "/" + subreads.subreads_id + "/" + subreads.subreads_id,
-                cores = limaCores,
-                dockerImage = dockerImages["lima"]
-        }
+    # Merge the bam files again.
+    call samtools.Merge as merge {
+        input:
+            bamFiles = ccs.ccsBam,
+            outputBamPath = subreadsName + ".ccs.bam"
+    }
 
-        scatter (bamFile in lima.limaBam) {
-            if (runIsoseq3Refine) {
-                String refineOutputPrefix = sub(basename(bamFile, ".bam"), "fl", "flnc")
-                call isoseq3.Refine as refine {
-                    input:
-                        inputBamFile = bamFile,
-                        primerFile = subreads.barcodes_file,
-                        outputDir = outputDirectory + "/" + subreads.subreads_id,
-                        outputNamePrefix = refineOutputPrefix,
-                        dockerImage = dockerImages["isoseq3"]
-                }
+    # Merge the report for MultiQC.
+    call mergePacBio as mergeCCSReport {
```

Comment:
Possibly provide the dockerImage here as well? To make sure the dockerImage can be changed through the Map[String, String] dockerImages input if necessary.

Reply:
I like it better when tasks just define their own docker image, otherwise you have to update in multiple places when you update to a more recent image. Users can still overwrite the docker image if they *really* want to via the workflow settings json.